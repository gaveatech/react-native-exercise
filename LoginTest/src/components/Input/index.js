import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

const Input = ({
  onChangeText,
  value,
  placeholder,
  keyboardType,
  password,
  autoFocus,
  error,
}) => {
  return (
    <InputWrapper>
      <TextBox
        {...{onChangeText, value, placeholder, keyboardType, autoFocus}}
        autoCorrect={false}
        secureTextEntry={password}
        testID="input"
      />
      {error && <Error>{error}</Error>}
    </InputWrapper>
  );
};

const TextBox = styled.TextInput`
  border: 1px solid #ccc;
  border-radius: 25px;
  padding: 10px 20px;
  min-width: 80%;
  font-family: 'RobotoRegular';
`;

const Error = styled.Text`
  font-size: 13px;
  font-family: 'RobotoThing';
  color: #ff0000;
  margin-left: 20px;
`;

const InputWrapper = styled.View`
  flex-direction: column;
  justify-content: flex-start;
  margin-bottom: 5px;
  height: 70px;
`;

TextBox.propTypes = {
  onChangeText: PropTypes.func.isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  keyboardType: PropTypes.string,
  password: PropTypes.bool,
  autoFocus: PropTypes.bool,
  error: PropTypes.string,
};

TextBox.defaultProps = {
  value: '',
  error: '',
  placeholder: '',
  keyboardType: 'default',
  password: false,
  autoFocus: false,
  onChangeText: () => null,
};

export default Input;
