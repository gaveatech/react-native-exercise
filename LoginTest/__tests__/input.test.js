import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react-native';

import Input from '../src/components/Input';

describe('<Input>', () => {
  it('Snapshot', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Add some text', () => {
    const TestFunc = jest.fn();
    const testText = 'Davi Camarinha';

    const {getByPlaceholderText} = render(
      <Input
        placeholder="Test Input"
        onChangeText={TestFunc}
        value={testText}
      />,
    );

    fireEvent.changeText(getByPlaceholderText('Test Input'), testText);
    expect(
      getByPlaceholderText('Test Input')._fiber.stateNode.props.value,
    ).toBe(testText);
  });
});
