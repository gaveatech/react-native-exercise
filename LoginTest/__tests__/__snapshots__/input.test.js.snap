// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> Snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
        "height": 70,
        "justifyContent": "flex-start",
        "marginBottom": 5,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    autoFocus={false}
    error=""
    keyboardType="default"
    onChangeText={[Function]}
    password={false}
    placeholder=""
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderColor": "#ccc",
          "borderRadius": 25,
          "borderStyle": "solid",
          "borderWidth": 1,
          "fontFamily": "RobotoRegular",
          "minWidth": "80%",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
        },
      ]
    }
    testID="input"
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
